ui/job-view/pushes/JobGroup.jsx

```diff
@@ -61,13 +60,10 @@ export class JobGroupComponent extends React.Component {
   };
 
   groupButtonsAndCounts(jobs, expanded) {
-    const {
-      selectedJob,
-      duplicateJobsVisible,
-      groupCountsExpanded,
-    } = this.props;
+    const { duplicateJobsVisible, groupCountsExpanded } = this.props;
     let buttons = [];
     const counts = [];
+    const selectedJobId = getSelectedJobId();
```

Comment:
We don't need to update this component when the ``selectedJob`` changes, just need to detect if there IS one in this function, so we can just check the URL.  This saves some re-rendering all over when we change jobs.